### PR TITLE
chore(capacity): organize integration test

### PR DESCRIPTION
**Pull request by enddynayn**
# Goal
Separate staking and using Capacity into
different files.

Added describe blocks to organize the test.

### CODE REVIEW

**Review by JoeCap08055 — APPROVED** [review]

File: integration-tests/capacity/staking.test.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        let stakeProviderId: u64;
+        before(async function () {
+            // Create and fund a keypair with accountBalance
+            // Use this keypair for stake operations
+            stakeKeys = createKeys("StakeKeys");
+            stakeProviderId = await createMsaAndProvider(stakeKeys, "StakeProvider", accountBalance);
+        });
+
+        it("successfully stakes the minimum amount", async function () {
+            await stakeToProvider(stakeKeys, stakeProviderId, 1n * cents);
```
[the remark this answers is not in the document]
Comment (enddynayn) [reply integration-tests/capacity/staking.test.ts]: done.

File: integration-tests/capacity/transactions.test.ts [hunk 2]
```diff
@@ -41,290 +30,195 @@ describe("Capacity Transaction Tests", function () {
...
+
+                const [grantDelegationEvent, chainEvents] = await grantDelegationOp.payWithCapacity();
+
+                if (grantDelegationEvent &&
+                    !(ExtrinsicHelper.api.events.msa.DelegationGranted.is(grantDelegationEvent))) {
+                    assert.fail("should return a DelegationGranted event");
+                }
+
+                // When Capacity has withdrawn an event CapacityWithdrawn is emitted.
+                if (chainEvents && chainEvents["capacity.CapacityWithdrawn"].isEmpty) {
```
Comment (mattheworris) [inline integration-tests/capacity/transactions.test.ts:82]:
```suggestion
  // Consider using this function from replenishment.ts (possibly moving it to helpers.ts)
  function assertEvent(events: EventMap, eventName: string) {
    assert(events.hasOwnProperty(eventName));
  }
  assertEvent(chainEvents, "capacity.CapacityWithdrawn");
```

File: integration-tests/capacity/transactions.test.ts [hunk 3]
```diff
@@ -41,290 +30,195 @@ describe("Capacity Transaction Tests", function () {
...
+                let delegatorKeys = createKeys("delegatorKeys");
+                let _delegatorMsaId = await ExtrinsicHelper.createMsa(delegatorKeys);
+
+                let providerKeys = createKeys("providerKeys");
+                let providerId = await createMsaAndProvider(providerKeys, "UnstakeProvider");
+
+                // Stake enough to cover the 2.8M grantDelegation cost
+                await assert.doesNotReject(stakeToProvider(stakeKeys, providerId, 3n * cents));
+
+                // grantDelegation costs about 2.8M and is the the capacity eligible
```
Comment (mattheworris) [inline integration-tests/capacity/transactions.test.ts:139]:
```suggestion
                // grantDelegation costs about 2.8M and is the capacity eligible
```

File: integration-tests/capacity/transactions.test.ts [hunk 4]
```diff
@@ -41,290 +30,195 @@ describe("Capacity Transaction Tests", function () {
...
+
+                const [grantDelegationEvent, chainEvents] = await grantDelegationOp.payWithCapacity();
+
+                if (grantDelegationEvent &&
+                    !(ExtrinsicHelper.api.events.msa.DelegationGranted.is(grantDelegationEvent))) {
+                    assert.fail("should return a DelegationGranted event");
+                }
+
+                // When Capacity has withdrawn an event CapacityWithdrawn is emitted.
+                if (chainEvents && chainEvents["capacity.CapacityWithdrawn"].isEmpty) {
```
Comment (mattheworris) [inline integration-tests/capacity/transactions.test.ts:157]:
```suggestion
                assertEvent(chainEvents, "capacity.CapacityWithdrawn");
                // See the previous comment for the function 
```

**Review by mattheworris — APPROVED** [review]
I like the new organization, much improved. I have suggested a few more opportunities for refactoring and some incorrect comments. 
I did not find any bugs, so I approved with some options for improvements, but not necessary to merge.

File: integration-tests/capacity/transactions.test.ts (hunk 2 above)
Comment (enddynayn) [reply integration-tests/capacity/transactions.test.ts:70]: Ill keep it in this file for now.

File: integration-tests/capacity/transactions.test.ts [hunk 5]
```diff
@@ -2,329 +2,209 @@ import "@frequency-chain/api-augment";
...
-        it("should fail to pay for a transaction with no msaId", async function () {
-            const payload = await generateDelegationPayload({
-                authorizedMsaId: delegatorProviderId,
-                schemaIds: [schemaId],
+
+            // A registered provider with Capacity but no tokens associated with the
+            // key should still be able to use polkadot UI to submit a capacity transaction.
+            // *All accounts will have at least an EXISTENTIAL_DEPOSIT = 1M.
+            // This test may still be valid if the txn cost is greater than 1M.
+            it("pays for a transaction with available capacity", async function () {
```
Comment (mattheworris) [inline integration-tests/capacity/transactions.test.ts:118]:
```suggestion
            it("pays for a transaction with available capacity and no free tokens", async function () {
```

File: integration-tests/capacity/staking.test.ts [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    devAccounts, createKeys, createMsaAndProvider,
+    stakeToProvider, fundKeypair,
+    getNextEpochBlock, TEST_EPOCH_LENGTH
+}
+    from "../scaffolding/helpers";
+import { firstValueFrom } from "rxjs";
+
+describe("Capacity Transaction Tests", function () {
+    let cents = 1000000n
+    let dollars = 100n * cents;
```
[the remark this answers is not in the document]
Comment (shannonwells) [reply integration-tests/capacity/staking.test.ts:16]: thanks for this

**Review by shannonwells — APPROVED** [review]
this is fine

**Pull request merged**